### PR TITLE
Fix Bucketize Conformance tests for Template plugin

### PR DESCRIPTION
**Pull request by nsemaev**
Taken SLT as a reference, data-size * 5 is the range for random values.
https://github.com/openvinotoolkit/openvino/blob/a883dc0b85e19a73a92e04f21e62af72bbeceea1/src/tests/functional/plugin/cpu/single_layer_tests/bucketize.cpp#L62-L68

### Details:
 - *Fix Bucketize Conformance tests inputs generation for Template plugin*

### Tickets:
 - *81520*

### CODE REVIEW

File: src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp [hunk 1]
```diff
@@ -486,8 +486,7 @@ ov::runtime::Tensor generate(const std::shared_ptr<ngraph::op::v3::Bucketize>& n
     InferenceEngine::Blob::Ptr blobPtr;
     switch (port) {
         case 0: {
-            unsigned long data_size = std::accumulate(begin(targetShape), end(targetShape), 1, std::multiplies<uint64_t>());
-            return create_and_fill_tensor(elemType, ov::Shape{data_size * 5}, 0, 10, 7235346);
+            return create_and_fill_tensor(elemType, targetShape, 2 * shape_size(targetShape), 0, 10, 7235346);
```
Comment (gkrivor) [inline src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:489]: 1st - I didn't get from a bug description - what are you fixing here? Could you describe original issue and way you'd chosen for the solution.
2nd - What does 2 * shape_size(targetShape) mean?

My thoughts:
I found only one implementation of create_and_fill_tensor in ov_tensor_utils.cpp, let's compare calls:
create_and_fill_tensor(element_type,        shape,              range,            start_from, resolution, seed) - definition
create_and_fill_tensor(elemType, ov::Shape{data_size * 5}, 0,                        10,        7235346) - original
create_and_fill_tensor(elemType, targetShape,   2 * shape_size(targetShape), 0,           10,    7235346) - your fix

I remember you said 723... magic number is a seed, but it isn't in existing implementation. Could it be an original root cause?

For me it looks like just a copy-paste typo while calling create_and_fill_tensor here (definition of *_unique_sequence doesn't have a "range" argument).

But regarding p.1 - I don't understand original issue. I suppose missing "range" in call. But also I don't fully understand why we call ov::Shape{data_size * 5} in original implementation (what does 5 mean?)...

Comment (nsemaev) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp]: Described the original problem and the method I chose to solve it in the description. Wrote another solution with a different method, looks more logical. Please give one more review

**Comment by nsemaev** [comment]
[report.xml.txt](https://github.com/openvinotoolkit/openvino/files/8315385/report.xml.txt)

File: src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp [hunk 2]
```diff
@@ -486,8 +486,7 @@ ov::runtime::Tensor generate(const std::shared_ptr<ngraph::op::v3::Bucketize>& n
     InferenceEngine::Blob::Ptr blobPtr;
     switch (port) {
         case 0: {
-            unsigned long data_size = std::accumulate(begin(targetShape), end(targetShape), 1, std::multiplies<uint64_t>());
-            return create_and_fill_tensor(elemType, ov::Shape{data_size * 5}, 0, 10, 7235346);
+            return create_and_fill_tensor_unique_sequence(elemType, targetShape, 0, 10, 7235346);
```
Comment (sbalandi) [inline src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:489]: How many dimensions does the targetShape have here for port 0 and 1 ?

Comment (nsemaev) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:489]: In both existing tests only one port (only port 0), its size 1*1

**Comment by nsemaev** [comment]
[report.xml.txt](https://github.com/openvinotoolkit/openvino/files/8346471/report.xml.txt)

**Review by gkrivor — APPROVED** [review]
I still against "magic numbers" in a source. Especially I confused by "* 5". But I don't think it is a blocker for merge because previous version of code has similar implementation.

**Review by sbalandi — APPROVED** [review]

**Pull request merged**